static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js

```diff
@@ -108,34 +108,34 @@ define([
-                        if (!outbrainBlockedByAds) {
-                            // if only a high priority ad on page then outbrain is merchandise compliant
-                            checkMediator.waitForCheck('isOutbrainMerchandiseCompliant').then(function (outbrainMerchandiseCompliant) {
-                                if (outbrainMerchandiseCompliant) {
-                                    module.load('merchandising');
-                                } else {
-                                    checkMediator.waitForCheck('isUserInNonCompliantAbTest').then(function (userInNonCompliantAbTest) {
-                                        userInNonCompliantAbTest ? module.load('nonCompliant') : module.load();
-                                    });
-                                }
-                            });
-                        }
-                    });
-                }
-            });
-        }
-
-        return Promise.resolve(true);
+        return checkMediator.waitForCheck('isOutbrainDisabled').then(function (outbrainDisabled) {
+            if (!outbrainDisabled) {
+                // if there is no merch component, load the outbrain widget right away
+                return loadInstantly().then(function(shouldLoadInstantly) {
+                    if (shouldLoadInstantly) {
+                        return checkMediator.waitForCheck('isUserInNonCompliantAbTest').then(function (userInNonCompliantAbTest) {
+                            userInNonCompliantAbTest ? module.load('nonCompliant') : module.load();
+                        });
+                    } else {
+                        // if a high priority merch component and low priority merch component on page block outbrain
+                        return checkMediator.waitForCheck('isOutbrainBlockedByAds').then(function(outbrainBlockedByAds) {
+                            if (!outbrainBlockedByAds) {
+                                // if only a high priority merch component on page then outbrain is merchandise compliant
+                                return checkMediator.waitForCheck('isOutbrainMerchandiseCompliant').then(function (outbrainMerchandiseCompliant) {
+                                    if (outbrainMerchandiseCompliant) {
+                                        module.load('merchandising');
+                                    } else {
+                                        return checkMediator.waitForCheck('isUserInNonCompliantAbTest').then(function (userInNonCompliantAbTest) {
+                                            userInNonCompliantAbTest ? module.load('nonCompliant') : module.load();
+                                        });
+                                    }
+                                });
+                            }
+                        });
+                    }
+                });
+            }
```

Comment:
The unit tests are the only place we chain a then to the outbrain.init and the function passed to the then is still executed whether or not we returned a resolved promise or not, however if it's better practise to explicitly return a resolved promise then I can do this.